source/standard_retry_strategy.c

```diff
@@ -248,11 +256,13 @@ static int s_standard_retry_acquire_token(
     return AWS_OP_SUCCESS;
 
 table_updated:
-    aws_hash_table_remove(&standard_strategy->token_buckets, &bucket_ptr->partition_id_cur, NULL, NULL);
+    aws_hash_table_remove(&standard_strategy->synced_data.token_buckets, &bucket_ptr->partition_id_cur, NULL, NULL);
```

Comment:
I think if we got here, then the mutex has already been unlocked and we're accessing the synced_data.

Reply:
yikes. good catch again. Also this was much simpler than I was making it.